resources/scripts/pytest_otel/Makefile

```
@@ -0,0 +1,113 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+PYTEST_OTEL_VERSION = 0.0.1
+PYTEST_OTEL_PKG = pytest-otel-$(PYTEST_OTEL_VERSION)
+VENV ?= .venv
+PYTHON ?= python3
+PIP ?= pip3
+GH_VERSION = 1.0.0
+
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Linux)
+	OS_FLAG := linux
+endif
+ifeq ($(UNAME_S),Darwin)
+	OS_FLAG := macOS
+endif
+UNAME_P := $(shell uname -m)
+ifeq ($(UNAME_P),x86_64)
+	ARCH_FLAG := amd64
+endif
+ifneq ($(filter %86,$(UNAME_P)),)
+	ARCH_FLAG := i386
+endif
+GH_BINARY = gh_$(GH_VERSION)_$(OS_FLAG)_$(ARCH_FLAG)
+GH = $(CURDIR)/bin/gh
+
+.SILENT:
+
+.PHONY: help
+help:
+	@echo "Targets:"
+	@echo ""
+	@grep '^## @help' Makefile|cut -d ":" -f 2-3|( (sort|column -s ":" -t) || (sort|tr ":" "\t") || (tr ":" "\t"))
+
+bin:
+	mkdir bin
+
+bin/gh: bin
+	curl -sSfL https://github.com/cli/cli/releases/download/v$(GH_VERSION)/$(GH_BINARY).tar.gz|tar xz
+	mv $(GH_BINARY)/bin/gh bin/gh
+	rm -fr $(GH_BINARY)
+
+## @help:virtualenv:Create a Python virtual environment.
+.PHONY: virtualenv
+virtualenv:
+	$(PYTHON) --version
+	test -d $(VENV) || virtualenv -q --python=$(PYTHON) $(VENV);\
+	source $(VENV)/bin/activate;\
+	$(PIP) install -q -r requirements.txt;
+
+## @help:install:Install APM CLI in a Python virtual environment.
+.PHONY: install
+install: virtualenv
+	source $(VENV)/bin/activate;\
+	$(PIP) install .;
+
+## @help:test:Run the test.
+.PHONY: test
+test: virtualenv install
+	OTEL_EXPORTER_OTLP_ENDPOINT=127.0.0.1:9444 \
```

```suggestion
	OTEL_EXPORTER_OTLP_ENDPOINT=127.0.0.1:4317 \
```

`4317` is the standard OTLP GRPC listen port